File: src/app/conjugation/forms/te-form-negative.ts
import {Conjugation, Word} from "../conjugation";
import {NonPastShortNegative} from "./non-past-short-negative";

export class TeFormNegative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {

        return new NonPastShortNegative().getConjugation(word)?.replaceLastKana('くて')
    }

    getTitle(): string {
        return "Te-Form, negative"
    }

    getSettingsTitle(): string {
        return "Negative"
    }

}

File: backend/app/conjugation/forms/te_form_negative.py
from ..core import Conjugation, Word
from .non_past_short_negative import NonPastShortNegative


class TeFormNegative(Conjugation):

    title = 'Te-Form, negative'
    settings_title = 'Negative'

    def conjugate(self, word: Word) -> Word | None:
        negative = NonPastShortNegative().conjugate(word)
        if negative is None:
            return None

        return negative.replace_last_kana('くて')
